landscaper/gardenlet/pkg/apis/imports/validation/validation.go

```diff
@@ -0,0 +1,129 @@
+func validateGardenletConfiguration(gardenletConfig *gardenletconfig.GardenletConfiguration, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if gardenletConfig.SeedSelector != nil {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("seedSelector"), "seed selector is forbidden, provide a seedConfig instead."))
+	}
+
+	if gardenletConfig.SeedConfig == nil {
+		return append(allErrs, field.Required(fldPath.Child("seedConfig"), "the seed configuration has to be provided. This is used to automatically register the seed."))
+	}
+
+	allErrs = append(allErrs, gardenletvalidation.ValidateGardenletConfiguration(gardenletConfig, fldPath)...)
+
+	if gardenletConfig.GardenClientConnection != nil && len(gardenletConfig.GardenClientConnection.Kubeconfig) > 0 {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("gardenClientConnection.kubeconfig"), "directly supplying a Garden kubeconfig and therefore not using TLS bootstrapping, is not supported."))
+	}
+
+	if gardenletConfig.SeedClientConnection != nil && gardenletConfig.SeedClientConnection.Kubeconfig != "" {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("seedClientConnection.kubeconfig"), "setting a Seed cluster kubeconfig is not supported. Instead, the landscaper creates a bootstrap kubeconfig with a bootstrap token as client credential for the Gardenlet."))
+	}
+
+	return allErrs
+}
+
+// validateBackup validates the Seed Backup configuration of the gardenlet landscaper imports
+func validateBackup(componentConfigSeedBackup *gardencore.SeedBackup, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if componentConfigSeedBackup == nil {
+		return append(allErrs, field.Required(fldPath, "the Seed has to to have backup enabled when the Gardenlet landscaper is configured to deploy a backup secret via \"seedBackupCredentials\""))
+	}
+
+	if len(componentConfigSeedBackup.Provider) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("provider"), "seed backup provider must be defined when configuring backups"))
+	}
+
+	if len(componentConfigSeedBackup.SecretRef.Name) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("secretRef.Name"), "seed backup secretRef must be set when configuring backups"))
+	}
+
+	return allErrs
+}
+
+// validateGardenletDeployment validates the deployment configuration of the landscaper gardenlet imports
+func validateGardenletDeployment(deployment *seedmanagementv1alpha1.GardenletDeployment, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if deployment.Image != nil {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("image"), "directly setting the image field is not supported, instead set the image via the component descriptor"))
+	}
+	toValidate := &seedmanagement.GardenletDeployment{}
+	if err := seedmanagementv1alpha1.Convert_v1alpha1_GardenletDeployment_To_seedmanagement_GardenletDeployment(deployment, toValidate, nil); err != nil {
```

Comment:
This will not be needed if you use the internal type `seedmanagement.GardenletDeployment` in your internal type as suggested in other comments.